### PR TITLE
DOC: fix formatting / links of API refs in 1.0.2 whatsnew

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
cc @TomAugspurger (did a few other clean-ups directly as well)

### CODE REVIEW

File: doc/source/whatsnew/v1.0.2.rst [hunk 1]
```diff
@@ -17,16 +17,17 @@ Fixed regressions
 
 - Fixed regression in :meth:`DataFrame.to_excel` when ``columns`` kwarg is passed (:issue:`31677`)
 - Fixed regression in :meth:`Series.align` when ``other`` is a DataFrame and ``method`` is not None (:issue:`31785`)
-- Fixed regression in :meth:`pandas.core.groupby.RollingGroupby.apply` where the ``raw`` parameter was ignored (:issue:`31754`)
-- Fixed regression in :meth:`pandas.core.window.Rolling.corr` when using a time offset (:issue:`31789`)
-- Fixed regression in :meth:`pandas.core.groupby.DataFrameGroupBy.nunique` which was modifying the original values if ``NaN`` values were present (:issue:`31950`)
+- Fixed regression in ``groupby(..).rolling(..).apply()`` (``RollingGroupby``) where the ``raw`` parameter was ignored (:issue:`31754`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.2.rst]: RollingGroupby is not included in our API docs, so can't link to  it

File: doc/source/whatsnew/v1.0.2.rst [hunk 2]
```diff
@@ -64,7 +65,6 @@ Bug fixes
 
 **Datetimelike**
 
-- Bug in :meth:`DataFrame.reindex` and :meth:`Series.reindex` when reindexing with a tz-aware index (:issue:`26683`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.2.rst]: moved to regressions, as this also fixed a regression (initially only a bug, but was backported later)

**Review by datapythonista — APPROVED** [review]
lgtm, thanks

**Comment by TomAugspurger** [comment]
Thakns.

**Comment by TomAugspurger** [comment]
I'll backport this manually.

**Pull request merged**